delta-app/src/main/java/io/cdap/delta/store/StateStore.java

```diff
@@ -95,8 +100,31 @@ public Long getLatestGeneration(String namespace, String pipelineName) throws IO
     return maxGeneration > 0 ? maxGeneration : null;
   }
 
+  public Collection<Integer> getWorkerInstances(DeltaPipelineId pipelineId) throws IOException {
```

Comment:
I might miss the write path. Do we have the logic of writing the worker instances to the file?

Reply:
yeah it's from changing the other methods in this class to take a DeltaWorkerId, and modify getPath() to include the instance id in the path.